numba/core/ir.py

```diff
@@ -543,6 +543,17 @@ def make_function(cls, name, code, closure, defaults, loc):
         op = 'make_function'
         return cls(op=op, name=name, code=code, closure=closure, defaults=defaults, loc=loc)
 
+    @classmethod
+    def null(cls, loc):
+        """
+        A node for null value.
+
+        This node is not handled by type inference. It is only added by
+        post-typing passes.
+        """
+        op = 'null'
+        return cls(op=op, loc=loc)
+
```

Comment:
Note to self: accidentally adding a `null` node into IR consumed by type inference will result in a:
```
Unsupported op-code encountered: null()
```

Reply:
Right, this is intentional as described in the docstring. This node is inserted post-inference by the ssa-stripping pass.

Reply:
My intention here was that I just wanted to see what happens if you make a mistake, essentially to check that the error points to the problem correctly, which it does.